pkg/splunk/enterprise/util.go

```diff
@@ -1578,3 +1618,59 @@ func validateMonitoringConsoleRef(c splcommon.ControllerClient, revised *appsv1.
 	//if the sts doesn't exists no need for any change
 	return nil
 }
+
+// getListOfClusterScopedApps returns the list of cluster scoped apps
+func getListOfClusterScopedApps(appDeployContext *enterpriseApi.AppDeploymentContext) []*enterpriseApi.AppDeploymentInfo {
+	scopedLog := log.WithName("getListOfClusterScopedApps")
+
+	var clusterScopedApps []*enterpriseApi.AppDeploymentInfo
+	for appSrcName, appSrcDeployInfo := range appDeployContext.AppsSrcDeployStatus {
+		deployInfoList := appSrcDeployInfo.AppDeploymentInfoList
+		for i := range deployInfoList {
+			appSrc, err := getAppSrcSpec(appDeployContext.AppFrameworkConfig.AppSources, appSrcName)
+			if err != nil {
+				// Error, should never happen
+				scopedLog.Error(err, "Unable to find App src", "App src name", appSrcName, "App name", deployInfoList[i].AppName)
+				continue
+			}
+
+			if appSrc.Scope == enterpriseApi.ScopeCluster && deployInfoList[i].DeployStatus != enterpriseApi.DeployStatusComplete {
+				clusterScopedApps = append(clusterScopedApps, &deployInfoList[i])
+			}
+		}
+	}
+
+	return clusterScopedApps
+}
+
+// areAllAppsCopiedToPod checks whether all the apps are copied to the target pod or not
+func areAllAppsCopiedToPod(appsList []*enterpriseApi.AppDeploymentInfo) bool {
+
+	if len(appsList) == 0 {
+		return false
+	}
+
+	for _, app := range appsList {
+		if app.PhaseInfo.Status != enterpriseApi.AppPkgPodCopyComplete {
+			return false
+		}
+	}
+	return true
+}
```

Comment:
As mentioned in my earlier comment, we have this info already with the cluster scoped apps tracking in place.